homeassistant/components/binary_sensor/mqtt.py

```diff
@@ -94,6 +94,8 @@ def state_message_received(topic, payload, qos):
                 self._state = True
             elif payload == self._payload_off:
                 self._state = False
+            else:  # Payload is not for this entity
+                return
```

Comment:
Add a warning that some things is going wrong with input

Reply:
Not sure what you mean by this. As I see it, it is possible to create several binary sensors (mqtt) that all listen to the same state_topic but for different payloads. 

Hence, the topic is correct, but the payload should only be handled by the entity where either payload_off or payload_on matches, else we do not want to update the state of the entity.

Reply:
Yeah, but we should log a warning, that he not found a match